src/main/scala/uk/gov/hmrc/ct/accounts/AC12.scala

```diff
@@ -38,17 +38,17 @@ case class AC12(value: Option[Int]) extends CtBoxIdentifier(name = "Current Turn
       boxRetriever.ac4()
   }
 
-  override def validate(boxRetriever: AccountsBoxRetriever with FilingAttributesBoxValueRetriever): Set[CtValidation] = {
+  override def validate(boxRetriever: AccountsBoxRetriever): Set[CtValidation] = {
       val errors = collectErrors(
         failIf(isFrs10xHmrcAbridgedReturnWithLongPoA(accountsStart, accountEnd)(boxRetriever)) {
           validateAsMandatory(this)
         },
-        failIf(boxRetriever.hmrcFiling().value)(
+        failIf(boxRetriever.filingAttributesBoxValueRetriever.hmrcFiling().value)(
           collectErrors(
             validateHmrcTurnover(boxRetriever, accountsStart, accountEnd)
           )
         ),
-        failIf(!boxRetriever.hmrcFiling().value && boxRetriever.companiesHouseFiling().value)(
+        failIf(!boxRetriever.filingAttributesBoxValueRetriever.hmrcFiling().value && boxRetriever.filingAttributesBoxValueRetriever.companiesHouseFiling().value)(
```

Comment:
You could pull `boxRetriever.filingAttributesBoxValueRetriever` out into a val to make these lines read a bit simpler